src/main/java/org/mule/extension/sftp/internal/connection/SftpClient.java

```diff
@@ -376,12 +389,29 @@ public void write(String path, InputStream stream, FileWriteMode mode) throws IO
    */
   public String getHome() {
     try {
-      return session.executeRemoteCommand("pwd").trim();
+      return executePWDCommandWithTimeout();
     } catch (IOException e) {
       throw new RuntimeException(e);
     }
   }
 
+  /**
+   * Apache Mina 2.9.2 - have a hardcoded timeout to infinite on executeRemoteCommand, if this method fails, the operation hangs.
+   * The solution at the moment is to add a timeout with the tools provided by mule sdk.
+   */
+  public String executePWDCommandWithTimeout() throws IOException {
+    try {
+      Scheduler getHomeScheduler =
+          schedulerService.cpuLightScheduler(SchedulerConfig.config().withShutdownTimeout(PWD_COMMAND_EXECUTION_TIMEOUT,
+                                                                                          PWD_COMMAND_EXECUTION_TIMEOUT_UNIT));
+      Future<String> submit = getHomeScheduler.submit(() -> session.executeRemoteCommand(PWD_COMMAND));
```

Comment:
@elcoya Do you know if the scheduler catches the InterruptedException that could throw the thread? It is not a good practice to ignore the InterruptedException doing nothing, but in this case, probably it is managed by the scheduler.

Reply:
I asked @GuidoGianfrate  to look in the runtime code but they don't cache the InterruptedException thrown by submit.get() either.

Reply:
Sounds about right. @GuidoGianfrate the InterruptedException woudn't be the same case as an illegal path. You could add it as a separate `catch` and throw a mule runtime exception.